node/syncblock.go

```diff
@@ -302,6 +302,61 @@ func (remoteNode *RemoteNode) GetBlocks(startHeight, endHeight uint32) ([]*block
 	return blocks, nil
 }
 
+// getNeighborsBlockHeaderByHeight returns the block header at a given height
+// from given neighbors by calling GetBlockHeaders on all of them concurrently.
+func (localNode *LocalNode) getNeighborsBlockHeaderByHeight(height uint32, neighbors []*RemoteNode) (*sync.Map, error) {
+	var allHeaders sync.Map
+	var wg sync.WaitGroup
+	for _, neighbor := range neighbors {
+		wg.Add(1)
+		go func(neighbor *RemoteNode) {
+			defer wg.Done()
+			headers, err := neighbor.GetBlockHeaders(height, height)
+			if err != nil {
+				log.Warningf("Get block header at height %d from neighbor %v error: %v", height, neighbor.GetID(), err)
+				return
+			}
+			allHeaders.Store(neighbor.GetID(), headers[0])
+		}(neighbor)
+	}
+	wg.Wait()
+	return &allHeaders, nil
+}
+
+// getNeighborsMajorityBlockHashByHeight returns the majority of given
+// neighbors' block hash at a give height
+func (localNode *LocalNode) getNeighborsMajorityBlockHashByHeight(height uint32, neighbors []*RemoteNode) common.Uint256 {
+	for i := 0; i < 3; i++ {
+		allHeaders, err := localNode.getNeighborsBlockHeaderByHeight(height, neighbors)
```

Comment:
sleep(RetryDelay) was cancelled?

Reply:
Yes I don't think it's necessary anymore